File: backend-java/src/main/java/com/biegsfritsstore/model/Boleta.java
package com.biegsfritsstore.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Document(collection = "boletas")
public class Boleta {

    @Id
    private String id;

    private String numero;
    private String clienteNombre;
    private String clienteWhatsapp;
    private List<BoletaItem> items;
    private double subtotal;
    private double igv;
    private double total;
    private String adminId;
    private LocalDateTime emitidaEn;

    @Data
    public static class BoletaItem {
        private String productId;
        private String nombre;
        private int cantidad;
        private double precioUnit;
        private double subtotal;
    }

    public Boleta() {
        this.emitidaEn = LocalDateTime.now();
    }
}

File: backend-node/models/Boleta.js
const mongoose = require('mongoose');

const BoletaItemSchema = new mongoose.Schema({
    productId: { type: String },
    nombre: { type: String, required: true },
    cantidad: { type: Number, required: true, min: 1 },
    precioUnit: { type: Number, required: true },
    subtotal: { type: Number, required: true }
}, { _id: false });

const BoletaSchema = new mongoose.Schema({
    numero: { type: String, required: true, unique: true },
    clienteNombre: { type: String, required: true },
    clienteWhatsapp: { type: String, default: '' },
    items: [BoletaItemSchema],
    subtotal: { type: Number, required: true },
    igv: { type: Number, required: true },
    total: { type: Number, required: true },
    adminId: { type: String, required: true },
    emitidaEn: { type: Date, default: Date.now }
}, {
    timestamps: true
});

module.exports = mongoose.model('Boleta', BoletaSchema);
